File: examples/src/Text/GetEffectiveValues.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using System
using System.Collections.Generic
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.text
{
    class GetEffectiveValues
    {
        public static void Run() {

            #ExStart:GetEffectiveValues
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Text()

            using (Presentation pres = new Presentation(dataDir + "Presentation1.pptx"))
            {
                shape = pres.slides[0].shapes[0] as IAutoShape

                ITextFrameFormat localTextFrameFormat = shape.text_frame.TextFrameFormat
                ITextFrameFormatEffectiveData effectiveTextFrameFormat = localTextFrameFormat.GetEffective()

                IPortionFormat localPortionFormat = shape.text_frame.paragraphs[0].portions[0].portion_format
                IPortionFormatEffectiveData effectivePortionFormat = localPortionFormat.GetEffective()
            }

            #ExEnd:GetEffectiveValues


        }
    }
}

File: examples/src/text/get_effective_values.py
import aspose.slides as slides


def get_effective_values(global_opts):
    with slides.Presentation(global_opts.data_dir + "text_add_animation_effect.pptx") as pres:
        shape = pres.slides[0].shapes[0]

        local_text_frame_format = shape.text_frame.text_frame_format
        effective_text_frame_format = local_text_frame_format.get_effective()

        local_portion_format = shape.text_frame.paragraphs[0].portions[0].portion_format
        effective_portion_format = local_portion_format.get_effective()
